res/decibel-mp.1
.TH DECIBEL-MP 1 "May 06, 2007"
.SH NAME
decibel-mp \- A Simple Music Player
.SH SYNOPSIS
.B Usage: decibel-mp [command] [arg1 arg2...]
.SH DESCRIPTION
\fBDecibel\fP is an audio player that aims at being very straightforward to use. It makes use of the GStreamer library to read audio files, and allows you to remote control using DBus.

.SH OPTIONS
.TP
\fB-Z, --remote-prev\fR
Jump to the previous track.
.TP
\fB-X, --remote-next\fR
Jump to the next track.
.TP
\fB-P, --remote-pause\fR
Pause or continue playing the current track.
.TP
\fB-H, --remote-shufl\fR
Shuffle the playlist.
.TP
\fB-C, --remote-clear\fR
Clear the playlist.
.TP
\fB-A, --remote-add=* [file1 file2...]\fR
Append the given files to the playlist.
.TP
\fB-S, --remote-set=* [file1 file2...]\fR
Set the playlist to the given files.
.TP
\fB-V, --remote-vol=* [0~100]\fR
Change volume to giving value.

.SH SEE ALSO
Homepage: http://decibel.silent-blade.org/
.SH AUTHOR
Decibel Audio Player was written by François Ingelrest <athropos@gmail.com>

This manual page was written by Emilio Pozuelo Monfort <pochu@ubuntu.com>.


ChangeLog
[+] New feature
[-] Bug fixed


v1.09 ()
    [-] Fixed help button not working in DesktopNotification module (#1018800)
    [ ] Don't set VBR information for MP3 files (#1202195)
    [ ] Don't manage any internal volume to avoid interfering with the system (#1202209)


v1.08 (19/09/11)
    [-] Keyboard interruption (Ctrl+C) doesn't work (#817803)
    [-] AudioScrobbler cache lost when Decibel is not closed properly (#787989)
    [-] Exception when the cover cannot be retrieved (#825083)
    [-] Album art extends past screen boundaries (#821452)
    [-] Remote sometimes fails with relative paths (#800813)
    [-] Make with DESTDIR writes incorrect path (#731383)
    [-] Folder listing mixes lower and upper case letters (#797862)
    [-] Skipping track just before the end skips the following track (#770346)
    [ ] Reverting playlist stops playback (#714548)
    [ ] Volume button not shown by default (#732956)
    [ ] On execution, raise if already running (#770111)
    [ ] Show filename instead of title when no title tag exists (#732583)
    [ ] Added a 'filename' column to the tracklist (right-click on column headers to display it)
    [ ] Convert from Libglade to GtkBuilder (#799738)


v1.07 (06/03/11)
    [+] Libraries may be filtered by genre using the popup menu
    [+] Can't play wav files (#653496)
    [+] Add an automatic shuffle feature (#510278)
    [+] Add a module that resumes playback on startup (#534031)
    [+] Added a 'lean' view mode (#523943)
    [+] Added a 'netbook' view mode (#711629)
    [+] Let external apps access the current cover (#715833)
    [-] Don't spawn multiple instances from the .desktop file (#704221)
    [ ] Playbin2 with pulseaudio sets volume to 100% (#511589)
    [ ] Make left clicks on slider act the same as middle clicks (#676548)
    [ ] "Add" option is missing on right-click (#677168)
    [ ] Don't preserve the ratio of covers when the distortion isn't noticeable


v1.06 (17/10/10)
    [+] Albums in the Library can be added to favorites using the popup menu
    [+] Added an option to search for covers in parent directories as well (#605644)
    [-] CD explorer not always removed when unloading the CD module (#603902)
    [-] AudioScrobbler submission blocked by '\0' bytes (#604016)
    [-] Tracklist scrolls up upon insertion (#607261)
    [-] List of explorers not kept sorted when renaming a library (#610202)
    [-] File chooser dialog box does not appear on some systems
    [-] Can't play previous track from the remote (#656040)
    [-] Scaling covers should preserve ratio (#654321)
    [ ] Revamped the preferences dialog box
    [ ] Respect the "Show hidden files" option when playing tracks from the File Explorer (#611703)
    [ ] Automatically show preferences the first time the application is started


v1.05 (27/06/10)
    [+] Stick to the Media Player Remote Interface Specifications (#534021)
    [+] Show some information on the selected tracks in the status bar
    [+] Added presets to the equalizer
    [+] [StatusIcon] Middle-click toggles pause, scroll changes volume (Laszlo Pandy)
    [+] Added Zeitgeist support (#579972)
    [-] Skips two songs when one is not found (#581654)
    [-] Fixed crash in some rare occasions when trying to set the cover for the current track
    [-] Fixed Gnome media keys no longer working when starting/closing multiple instances
    [-] Fixed labels not always properly updated in the equalizer
    [ ] Ignore disc number when it looks like '1/1' (#596350)
    [ ] Improved startup time
    [ ] The playbin2 GStreamer component is now used by default (use the --playbin flag to use the old playbin component)
    [ ] Equalizer module has a menu item (CTRL-E) when enabled (#495761)
    [ ] Close properly on CTRL-C


v1.04 (13/05/10)
    [+] Show album cover in desktop notifications if both modules are enabled (#418076)
    [+] Added a wildcard (*) for cover names (#541763)
    [+] Added more keyboard shortcuts to the tracklist (#523689)
    [+] CD-ROM read speed can be configured (#502297)
    [+] Added a bitrate column in the tracklist (hidden by default)
    [-] Drag'n'Drop into the playlist not always working correctly (#501114)
    [-] Improper sorting when adding tracks that don't all have a performer tag (#509494)
    [-] Don't allow playback of a disc when there is no disc in the drive
    [-] Removing a root folder in the File Explorer doesn't work correctly (#496154)
    [-] Cannot open playlists using backslashes as path separators (#526273)
    [-] Cannot play files with a percent sign in their path (#528442)
    [-] Cannot start when view mode was set to nano in v1.02 (#512892)
    [-] [AudioScrobbler] Song not submitted when quitting while playing (#565828)
    [-] Fixed passwords not being stored on Ubuntu 10.04 (#578124)
    [-] Time slider sometimes stops moving (#424976)
    [-] Fixed lowercase/uppercase issue in the library (#513965)
    [ ] Gloss effect on covers can be disabled by using the --no-glossy-cover option on the command line (#562905)
    [ ] "Save as" dialog now opens the last directory used (#523692)
    [ ] Don't start a new instance when there is already one running (#490719)
    [ ] Play button plays selected track if any (#511720)


v1.03 (10/01/10)
    [+] Improved the look of the cover thumbnail
    [-] Fixed grey text not always visible once selected (#331867)
    [-] Fixed removing a root folder in the File Explorer not working correctly (#496154)
    [ ] Merged mini and nano modes
    [ ] Fixed deprecation warnings about the tooltips


v1.02 (29/11/09)
    [+] Added support for .ape files (#425326)
    [+] In the list of explorers, replace "Audio CD" by the actual disc name when known (#425768)
    [+] Added support for ac3 and mp2 files (#442843)
    [+] Added an option to the File Explorer to add files to the playlist by their name (#311427)
    [-] Fixed files with a '#' character in their path not being played (#428307)

res/decibel-mp.sh
#!/usr/bin/env sh
# Author: Ingelrest François (Francois.Ingelrest@gmail.com)
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Library General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA

HOMEURL="@HOMEURL@"
APPNAME="@APPNAME@"
DATADIR="@DATADIR@"
VERSION="@VERSION@"

PYDIR="$DATADIR/$APPNAME"

p_ctrl=''
p_list=''

for i in "$@"; do
  case $i in
    --help | -h)
      echo ""
      echo "Usage: decibel-mp [options] [FILE(s)]"
      echo ""
      exit
      ;;
    --version | -v)
      echo "$APPNAME-$VERSION"
      exit
      ;;
    --remote-prev  | -Z  ) p_ctrl=PREV  ;;
    --remote-next  | -X  ) p_ctrl=NEXT  ;;
    --remote-pause | -P  ) p_ctrl=PAUSE ;;
    --remote-clear | -C  ) p_ctrl=CLEAR ;;
    --remote-shufl | -H  ) p_ctrl=SHUFFLE ;;
    --remote-vol=* | -V=*) p_ctrl=VOLUME; p_list=${i#*=} ;;
    --remote-add=* | -A=*) p_ctrl=ADD   ; p_list=${i#*=} ;;
    --remote-set=* | -S=*) p_ctrl=SET   ; p_list=${i#*=} ;;
  esac
done

if [ ! -z $p_ctrl ]; then
  python3 "$PYDIR/remote-control.py" "$p_ctrl" "$p_list"
else
  python3 "$PYDIR/player-instance.py"
fi
